mbuild/compound.py

```diff
@@ -254,15 +254,26 @@ def visualize(self, show_ports=False):
             __IPYTHON__
         except NameError:
             import tempfile
-            f = tempfile.NamedTemporaryFile(prefix='visualize_{}'.format(self.__class__.__name__), suffix='.mol2')
-            filename = f.name
-            print('vis: ', show_ports)
-            self.save(filename, show_ports=show_ports)
-            try:
-                os.system('vmd {}'.format(filename))
-            except OSError:
-                print("Visualization with VMD failed. Make sure it is installed"
-                      "correctly and launchable from the command line via 'vmd'.")
+            if sys.platform.startswith('win'):
+                f = tempfile.mkstemp(suffix='.mol2')
+                filename = f[1]
```

Comment:
I would just write `filename = tempfile.mkstemp(suffix='.mol2')[1]` to avoid the extra variable. Same in lin 269